driver/ppm_fillers.c

```diff
@@ -5718,8 +5719,8 @@ int f_sys_dup3_x(struct event_filler_arguments *args)
 	/*
 	 * flags
 	 */
-	syscall_get_arguments_deprecated(args, 2, 1, &val);
-	res = val_to_ring(args, dup3_flags_to_scap(val), 0, false, 0);
+	syscall_get_arguments_deprecated(args, 2, 1, &flags);
```

Comment:
unfortunately here you have to keep the `&val` because you need to read 8 bytes so an `unsigned long` is you use `&flags` this will cause a `write_overflow`. probably you need to some something like
```c
syscall_get_arguments_deprecated(args, 2, 1, &val);
flags = val;
```

Reply:
Great catch @Andreagit97!

Reply:
Ok, I left the original call and just did a cast on val when passed to dup3_flags_to_scap